polynote-frontend/polynote/ui/component/notebook/comment.ts

```diff
@@ -87,7 +87,7 @@ export class CommentHandler extends Disposable {
                    if (maybeRoot && maybeRoot.uuid === commentId) {
                        // If this is a root comment, we delete it and it's children
                        // First, the children
-                       Object.keys(maybeRoot.rootChildren(currentComments)).forEach(commentId => allCommentsState.update(() => removeKey(commentId)))
+                       maybeRoot.rootChildren(currentComments).forEach(childComment => allCommentsState.update(() => removeKey(childComment.uuid)))
```

Comment:
What was the reason for this change? (it's a fine change, just wondering whether it does anything I'm not seeing)

Reply:
the implementation of CommentRoot.rootChildren() returns an array of CellComments indexed with integers (so the first comment has key 0, the next has key 1 etc). so previously, this line (90) was trying to remove comments with ID 0, 1, etc, which don't exist, so if you deleted a root comment that had a reply, both comment and reply would 'disappear' but the child comment wouldn't get deleted and would pop back up if another comment was made on the same selection.

Reply:
aha! yikes 🤦